### PR TITLE
Add Monthly to Plans Grid

### PR DESCRIPTION
**Pull request by claudiucelfilip**
#### Proposed Changes

* This adds the monthly toggle for the Plans grid on the Signup flow and Plans page.

Screenshot on 2022-06-30 at 17-32-40
Screenshot on 2022-06-30 at 17-31-36

#### TODOs
- [x] recheck the monthly Pro plan once the backend work is done
- [ ] the discount is disabled by passing an empty `plans` array to the `IntervalTypeToggle`. Will probably need to pass the correct plan slugs and make sure the calculations work after the prices are fetched from the backend.
- [x] show the toggle behind an experiment check
- [x] fix failing tests
- [ ] (Optional) use strikethrough text for the annual plan only features and mention they're available on the annual plan.
- [ ] (Optional) use a better way to show the Starter Monthly plan is unavailable.
- [ ] (Optional) use the same width for the toggle (now the toggle Plans page is wider) 

#### Testing Instructions

* Make sure your sandbox is up to date.
* Apply D83557-code to your sandboxed `public-api.wordpress.com`.
  * You may need to `rebase` the arcpatched branch to get the experiment working.
* Assign yourself to the `treatment` group of the `calypso_show_interval_type_selector_2022_07` experiment.
* Create a site and you will see the interval type selector on the plans step.
* There will be the same selector on the My Site plans page.
  image
* Pick the monthly Pro plan and the purchase should be successfully processed.

Related to #

### CODE REVIEW

File: client/my-sites/plans-comparison/plans-comparison.tsx [hunk 1]
```diff
@@ -478,6 +481,7 @@ export const PlansComparison: React.FunctionComponent< Props > = ( {
 		<>
 			<Global styles={ globalOverrides } />
 			<QueryProductsList />
+
```
Comment (chriskmnds) [inline client/my-sites/plans-comparison/plans-comparison.tsx]:
```suggestion
```

File: client/my-sites/plans-features-main/plan-type-selector.tsx [hunk 2]
```diff
@@ -162,6 +163,27 @@ export const IntervalTypeToggle: React.FunctionComponent< IntervalTypeProps > =
...
 };
 
+export const ExperimentalIntervalTypeToggle: React.FunctionComponent< IntervalTypeProps > = (
+	props
+) => {
+	return (
+		<ProvideExperimentData name="calypso_show_interval_type_selector_2022_07">
+			{ ( isLoading, experimentAssignment ) => {
+				if ( isLoading || ! props.intervalType ) {
+					return <></>;
```
Comment (chriskmnds) [inline client/my-sites/plans-features-main/plan-type-selector.tsx:173]: I think `return null` here would also be correct?

File: client/my-sites/plans-features-main/plan-type-selector.tsx [hunk 3]
```diff
@@ -162,6 +163,27 @@ export const IntervalTypeToggle: React.FunctionComponent< IntervalTypeProps > =
...
+				}
+
+				const isTreatment = 'treatment' === experimentAssignment?.variationName;
+				if ( ! isTreatment ) {
+					return <></>;
+				}
+
+				return <IntervalTypeToggle { ...props } />;
```
Comment (chriskmnds) [inline client/my-sites/plans-features-main/plan-type-selector.tsx]: I may be missing something, but we only show the toggle when the assignment is `treatment` right? Maybe we can simplify like below in that case? (not tested) 🤔 

```
if ( 'treatment' === experimentAssignment?.variationName ) {
    return <IntervalTypeToggle { ...props } />;
}

return null;
```

File: client/signup/steps/plans/index.jsx [hunk 4]
```diff
@@ -174,11 +175,20 @@ export class PlansStep extends Component {
 		if ( eligibleForProPlan ) {
 			const selectedDomainConnection =
 				this.props.progress?.domains?.domainItem?.product_slug === 'domain_map';
+			const intervalType = this.getIntervalType();
 			return (
 				<div>
 					{ errorDisplay }
+					<ExperimentalIntervalTypeToggle
+						experiment="calypso_plugins_search_page_jetpack_search"
```
Comment (chriskmnds) [inline client/signup/steps/plans/index.jsx]: Not sure if this prop is needed. It doesn't look like it's used anywhere?

File: client/my-sites/plans-comparison/plans-comparison.tsx [hunk 5]
```diff
@@ -488,30 +492,35 @@ export const PlansComparison: React.FunctionComponent< Props > = ( {
...
-								translate={ translate }
-							>
-								{ selectedDomainConnection && <PlansDomainConnectionInfo plan={ plan } /> }
-								<PlansComparisonAction
-									currentSitePlanSlug={ sitePlan?.product_slug }
+						{ plans.map( ( plan, index ) => {
+							const isDomainConnectionDisabled =
+								selectedDomainConnection && [ TYPE_FREE, TYPE_FLEXIBLE ].includes( plan.type );
+							const isMonthlyPlan = plan?.term === TERM_MONTHLY;
+							const isMonthlyPlanDisabled = 'monthly' === intervalType && ! isMonthlyPlan;
```
Comment (chriskmnds) [inline client/my-sites/plans-comparison/plans-comparison.tsx:498]: I wish we wouldn't be doing this logic inline the JSX tree this way. It's almost like a call for an abstraction/component that's missing. No need to worry if anything complex to address.

File: client/my-sites/plans-comparison/plans-comparison.tsx [hunk 6]
```diff
@@ -488,30 +492,35 @@ export const PlansComparison: React.FunctionComponent< Props > = ( {
...
-										selectedDomainConnection && [ TYPE_FREE, TYPE_FLEXIBLE ].includes( plan.type )
-									}
-									onClick={ () => onSelectPlan( planToCartItem( plan ) ) }
-								/>
-							</PlansComparisonColHeader>
-						) ) }
+									currencyCode={ currencyCode }
+									price={ prices[ index ].price }
+									originalPrice={ prices[ index ].originalPrice }
+									translate={ translate }
```
Comment (chriskmnds) [inline client/my-sites/plans-comparison/plans-comparison.tsx:507]: I guess we don't need to pass `translate` as a prop? Can't `PlansComparisonColHeader` call `useTranslate`?

File: client/my-sites/plans-comparison/plans-comparison-row.tsx [hunk 7]
```diff
@@ -78,10 +78,14 @@ export const PlansComparisonRow: React.FunctionComponent< Props > = ( {
...
 			/>
 			{ plans.map( ( plan ) => {
-				const includedFeature = intersection(
-					plan.getPlanCompareFeatures?.() || [],
-					feature.features
-				)[ 0 ];
+				let planCompareFeatures = plan.getPlanCompareFeatures?.() || [];
+
+				if ( PLAN_WPCOM_PRO_MONTHLY === plan.getStoreSlug() ) {
+					const annualPlanFeatures = plan.getAnnualPlansOnlyFeatures?.() || [];
```
Comment (chriskmnds) [inline client/my-sites/plans-comparison/plans-comparison-row.tsx:84]: This was a bit hard to grasp seeing `getAnnualPlansOnlyFeatures` on a plan instance of "monthly". Maybe if this was a separate utility method that's not attached to the plan object would have helped. No need to worry if anything complex to address.

**Review by chriskmnds — COMMENTED** [review]
Tests out as advertised. I left some comments, but nothing critical I believe. 

Design-wise I think accessibility/clarity could have been a little better here. The transition/differences between the two options could have been more obvious maybe or highlighted better. "custom domain name" (only feature different it looks like) may be hidden from view; the transition from "checkmark" to "cross" is also kinda easy to miss. It's almost like we expect the user to do this over and over and study the list really closely.

Kapture 2022-07-12 at 15 13 04

cc @lucasmendes-design

**Comment by lucasmendes-design** [comment]
Thanks for pinging me @chriskmnds. 

Do we have time to change that? I think we can do a small change in the layout to better solve that problem. In addition, I think it's better in terms of layout as well. Wdyt?

Sorry for this, I should have shared that UI. 

**Web**
image

**Mobile** 
image

**Comment by lucasmendes-design** [comment]
btw, could we change the title position?

**Comment by chriskmnds** [comment]
> Do we have time to change that?

I don't know, but the mockups you share push "custom domain name" further up, which is definitely better to not accidentally miss. Not sure about the toggle in the side column though.

So my point was about the grid (and information) reacting to the selection. Obviously, it's a significant change (no more "custom domain name") and it felt a little obscured. It might also help if we had some transition/effect on that field to highlight it better (something like gray it out and transition it back maybe - I'm not an expert, but I think I've seen similar behavior elsewhere), ~or adding an extra notification somewhere~ (probably too much if it's only for something [not being free](https://github.com/Automattic/wp-calypso/pull/65126#issuecomment-1182519688)). 

p.s. you can also end up in checkout with a paid domain and the monthly plan, and nothing further mentions that you can't use that domain with the plan you are about to pay for. Is my understanding correct though (that "custom domain name" means any paid domain)?

Screenshot 2022-07-13 at 12 12 06 AM

(sorry if these are minor points and I'm making a fuss out of)

**Comment by chriskmnds** [comment]
> p.s. you can also end up in checkout with a paid domain and the monthly plan, and nothing further mentions that you can't use that domain with the plan you are about to pay for. Is my understanding correct though (that "custom domain name" means any paid domain)?

Ok - I think I got this part wrong. The two are unrelated, right? (i.e. custom domain marked as "not included" only means it's not free with the plan, not that you can't purchase one separately and use it with the site)

**Review by chriskmnds — APPROVED** [review]
If anything from @lucasmendes-design's designs to address or the code comments, otherwise didn't notice anything critical while testing.

**Comment by gmovr** [comment]
One small feedback ref convo: p1657704116350419-slack-C031TFM2NKC 

If I'm on a Starter plan (yearly) I shouldn't see the monthly toggle, as I'm not technically able to upgrade from Starter to monthly Pro. :) 

Yearly plans shouldn't see monthly.
Screenshot 2022-07-13 at 11 20 02

**Comment by lucasmendes-design** [comment]
The images that I shared were just about the Monthly and Annually feature, and not about the feature order. Sorry for the confusion. I circulated in red. 

image
image

> If I'm on a Starter plan (yearly) I shouldn't see the monthly toggle, as I'm not technically able to upgrade from Starter to monthly Pro. :)

Let me see if I understand, to upgrade from Starter (yearly) to Pro (Monthly). Users need to cancel their Starter Plans (yearly) and then purchase Pro (Monthly), am I right? @gmovr 

>  Is my understanding correct though (that "custom domain name" means any paid domain)?

AFAIK, yes, it's the same thing. 

> you can also end up in checkout with a paid domain and the monthly plan, and nothing further mentions that you can't use that domain with the plan you are about to pay for. 

I don't know if this is accurate. I think users can not make this domain primary but they can map the domain to their websites. (redirect).

**Comment by gmovr** [comment]
Regarding domains, you just won't get a free domain with monthly plans. 

You can still map a domain you own, and set it as your primary domain -- or choose to buy a domain via WordPress.com. Either will work, it's just not free/included except for yearly plans.
Screenshot 2022-07-13 at 15 37 37
 :)

File: client/my-sites/plans-features-main/plan-type-selector.tsx (hunk 2 above)
Comment (taggon) [reply client/my-sites/plans-features-main/plan-type-selector.tsx:173]: I would use it if the type checker didn't compain. `ProvideExperimentData` doesn't allow `null` as a return value.

File: client/my-sites/plans-features-main/plan-type-selector.tsx (hunk 3 above)
Comment (taggon) [reply client/my-sites/plans-features-main/plan-type-selector.tsx]: Again, `null` is not allowed here. :)

File: client/signup/steps/plans/index.jsx (hunk 4 above)
Comment (taggon) [reply client/signup/steps/plans/index.jsx]: Oops. The new component used to take `experiment` prop which doesn't exist. Wiped out. Thanks. :)

File: client/my-sites/plans-comparison/plans-comparison.tsx (hunk 6 above)
Comment (taggon) [reply client/my-sites/plans-comparison/plans-comparison.tsx:507]: @chriskmnds The component, `PlansComparisonColHeader` was intentionally designed as a stateless component. That's why the translate is passed as a prop here. You may find many other components that take `translate` prop for the same purpose. :)

**Comment by taggon** [comment]
With the latest commit, the interval type toggle will be hidden from Starter users.

**Comment by gmovr** [comment]
Screenshot 2022-07-14 at 11 56 43
Noticed one thing more! It's minor, but with monthly there's no 24/7 chat support. On monthly plans we don't have live chat. :)

**Pull request merged**